common/web/keyboard-processor/src/text/outputTarget.ts

```diff
@@ -439,12 +439,14 @@ namespace com.keyman.text {
         if(dn > this.caretIndex) {
           dn = this.caretIndex;
         }
+        this.adjustDeadkeys(-dn);
         this.text = this.text.kmwSubstr(0, this.caretIndex - dn) + this.getTextAfterCaret();
         this.caretIndex -= dn;
       }
     }
 
     insertTextBeforeCaret(s: string): void {
+      this.adjustDeadkeys(s._kmwLength());
```

Comment:
Compare:

https://github.com/keymanapp/keyman/blob/1a7ad09fdb3e14a6d117e8555775745bece555d8/web/source/dom/targets/touchAlias.ts#L53-L67

Reply:
Why `s._kmwLength()` as opposed to `s.kmwLength()`?

Reply:
Most of the calls in that file use `_kmwLength`, so it follows the same pattern.